src/org/labkey/test/tests/DataClassFolderExportImportTest.java

```diff
@@ -0,0 +1,351 @@
+package org.labkey.test.tests;
+
+import org.junit.BeforeClass;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.labkey.test.BaseWebDriverTest;
+import org.labkey.test.Locator;
+import org.labkey.test.TestFileUtils;
+import org.labkey.test.TestTimeoutException;
+import org.labkey.test.categories.DailyC;
+import org.labkey.test.components.CustomizeView;
+import org.labkey.test.params.FieldDefinition;
+import org.labkey.test.params.experiment.DataClassDefinition;
+import org.labkey.test.util.DataRegionTable;
+import org.labkey.test.util.LogMethod;
+import org.labkey.test.util.PortalHelper;
+import org.labkey.test.util.TestDataGenerator;
+import org.labkey.test.util.exp.DataClassAPIHelper;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.hasItems;
+import static org.junit.Assert.*;
+
+@Category({DailyC.class})
+public class DataClassFolderExportImportTest extends BaseWebDriverTest
+{
+    private static final String PROJECT_NAME = "DataClassExportFolderTest";
+    private static final String IMPORT_PROJECT_NAME = "DataClassImportFolderTest";
+
+    private final File DATAFILE_DIRECTORY = TestFileUtils.getSampleData("fileTypes");
+    private final File SAMPLE_CSV = new File(DATAFILE_DIRECTORY, "csv_sample.csv");
+    private final File SAMPLE_JPG = new File(DATAFILE_DIRECTORY, "jpg_sample.jpg");
+    private final File SAMPLE_PDF = new File(DATAFILE_DIRECTORY, "pdf_sample.pdf");
+    private final File SAMPLE_TIF = new File(DATAFILE_DIRECTORY, "tif_sample.tif");
+    private List<File> _attachments = List.of(SAMPLE_CSV, SAMPLE_JPG, SAMPLE_PDF, SAMPLE_TIF);
+
+    @Override
+    protected void doCleanup(boolean afterTest) throws TestTimeoutException
+    {
+        super.doCleanup(afterTest);
+    }
+
+    @BeforeClass
+    public static void setupProject()
+    {
+        DataClassFolderExportImportTest init = (DataClassFolderExportImportTest) getCurrentTest();
+
+        init.doSetup();
+    }
+
+    private void doSetup()
+    {
+        // Delete the import project if it exists.
+        _containerHelper.deleteProject(IMPORT_PROJECT_NAME, false);
+        // create the import project because we'll need it
+        _containerHelper.createProject(IMPORT_PROJECT_NAME);
+
+        PortalHelper portalHelper = new PortalHelper(this);
+        _containerHelper.createProject(PROJECT_NAME);
+
+        projectMenu().navigateToProject(PROJECT_NAME);
+        portalHelper.addWebPart("Data Classes");
+        portalHelper.addWebPart("Sample Types");
+        portalHelper.addWebPart("Experiment Runs");
```

Comment:
Group multiple `addWebPart` calls to avoid entering/exiting page admin mode repeatedly.
```suggestion
        portalHelper.doInAdminMode(() -> {
            portalHelper.addWebPart("Data Classes");
            portalHelper.addWebPart("Sample Types");
            portalHelper.addWebPart("Experiment Runs");
        });
```